packages/playwright-core/src/utils/processLauncher.ts

```diff
@@ -190,14 +190,23 @@ export async function launchProcess(options: LaunchProcessOptions): Promise<Laun
     }
   }
 
+  function sleep(ms: Number) {
+    childProcess.spawnSync(process.execPath, ['-e', 'setTimeout(function(){},' + ms + ')']);
+  }
+
   function killProcessAndCleanup() {
     killProcess();
     options.log(`[pid=${spawnedProcess.pid || 'N/A'}] starting temporary directories cleanup`);
     for (const dir of options.tempDirectories) {
-      try {
-        rimraf.sync(dir, { maxBusyTries: 10 });
-      } catch (e) {
-        options.log(`[pid=${spawnedProcess.pid || 'N/A'}] exception while removing ${dir}: ${e}`);
+      for (let i = 0; i < 10; i++) {
+        try {
+          // maxBusyTries has bugs
```

Comment:
It does not have bugs, but it only works in async mode, so the code here was simply wrong. Doing it 10 times synchronously does not help though, we need a proper async solution.

Reply:
Changed from async to sync in the last commit.
https://github.com/microsoft/playwright/commit/e8c512dbeb3f69ba8dde7fc22d7ee27843f1629a
I don't know the full logic, for smaller changes I chose this solution.